pkg/distributor/push.go

```diff
@@ -190,9 +226,18 @@ func toHTTPStatus(ctx context.Context, pushErr error, limits *validation.Overrid
 	return http.StatusInternalServerError
 }
 
-func addHeaders(w http.ResponseWriter, err error) {
+func addHeaders(w http.ResponseWriter, err error, r *http.Request, code int, retryCfg *RetryConfig, logger glog.Logger) {
 	var doNotLogError middleware.DoNotLogError
 	if errors.As(err, &doNotLogError) {
 		w.Header().Set(server.DoNotLogErrorHeaderKey, "true")
 	}
+
+	if (code == http.StatusTooManyRequests || code/100 == 5) && retryCfg != nil {
+		var retrySeconds string
+		if retryCfg.Enabled {
+			retrySeconds = calculateRetryAfter(r, retryCfg)
+			w.Header().Set("Retry-After", retrySeconds)
+		}
+		logger.Log("msg", "set retry-after", "enabled", retryCfg.Enabled, "retry-after", retrySeconds, "maxAllowedAttemps", retryCfg.MaxAllowedAttempts)
```

Comment:
No need to log this.

Reply:
i'm ok with removing logging if there's a tag on the trace at least